src/search.js

```diff
@@ -45,28 +45,40 @@ module.exports = function(list) {
   };
   var search = {
     list: function() {
+      // Extract quoted phrases "word1 word2" from original searchString
+      // searchString is converted to lowercase by List.js
+      var words = [], phrase, ss = searchString;
+      while ((phrase = ss.match(/"([^"]+)"/)) !== null) {
+        words.push(phrase[1]);
+        ss = ss.substring(0,phrase.index) + ss.substring(phrase.index+phrase[0].length);
+      };
+      // Get remaining space-separated words (if any)
+      ss = ss.trim();
+      if (ss.length) words = words.concat(ss.split(/\s+/));
       for (var k = 0, kl = list.items.length; k < kl; k++) {
-        search.item(list.items[k]);
-      }
-    },
-    item: function(item) {
-      item.found = false;
-      for (var j = 0, jl = columns.length; j < jl; j++) {
-        if (search.values(item.values(), columns[j])) {
-          item.found = true;
-          return;
-        }
-      }
-    },
-    values: function(values, column) {
-      if (values.hasOwnProperty(column)) {
-        text = list.utils.toString(values[column]).toLowerCase();
-        if ((searchString !== "") && (text.search(searchString) > -1)) {
-          return true;
+        var item = list.items[k];
+        item.found = false;
+        if (!words.length) continue;
+        for (var i = 0, il = words.length; i < il; i++) {
+          var word_found = false;
+          for (var j = 0, jl = columns.length; j < jl; j++) {
+            var values = item.values(), column = columns[j];
```

Comment:
Lift `var values = item.values()`out of the two inner loops?

Reply:
Yes, definite improvement :) Thanks